mcs/class/Mono.CSharp/monotouch.cs

```diff
@@ -28,797 +28,83 @@
-			}
-		}
-
-		public override Type PropertyType {
-			get {
-				throw new NotSupportedException ();
-			}
-		}
-
-		#endregion
-
-		public MethodToken GetToken()
-		{
-			throw new NotSupportedException ();
-		}
-
-		public void SetGetMethod (object arg)
-		{
-			throw new NotSupportedException ();
-		}
-
-		public void SetSetMethod (object arg)
-		{
-			throw new NotSupportedException ();
-		}
-
-		public void SetCustomAttribute (params object[] args)
+		public static MethodToken GetToken (this MethodBuilder builder)
 		{
 			throw new NotSupportedException ();
 		}
 	}
 
-	public class FieldBuilder : FieldInfo
+	static class ModuleBuilderExtensions
 	{
-		#region implemented abstract members of MemberInfo
-
-		public override bool IsDefined (Type attributeType, bool inherit)
-		{
-			throw new NotSupportedException ();
-		}
-
-		public override object[] GetCustomAttributes (bool inherit)
-		{
-			throw new NotSupportedException ();
-		}
-
-		public override object[] GetCustomAttributes (Type attributeType, bool inherit)
-		{
-			throw new NotSupportedException ();
-		}
-
-		public override Type DeclaringType {
-			get {
-				throw new NotSupportedException ();
-			}
-		}
-
-		public override string Name {
-			get {
-				throw new NotSupportedException ();
-			}
-		}
-
-		public override Type ReflectedType {
-			get {
-				throw new NotSupportedException ();
-			}
-		}
-
-		#endregion
-
-		#region implemented abstract members of FieldInfo
-
-		public override object GetValue (object obj)
-		{
-			throw new NotSupportedException ();
-		}
-
-		public override void SetValue (object obj, object value, BindingFlags invokeAttr, Binder binder, System.Globalization.CultureInfo culture)
-		{
-			throw new NotSupportedException ();
-		}
-
-		public override FieldAttributes Attributes {
-			get {
-				throw new NotSupportedException ();
-			}
-		}
-
-		public override RuntimeFieldHandle FieldHandle {
-			get {
-				throw new NotSupportedException ();
-			}
-		}
-
-		public override Type FieldType {
-			get {
-				throw new NotSupportedException ();
-			}
-		}
-
-		#endregion
-
-		public void SetConstant (object arg)
-		{
-			throw new NotSupportedException ();
-		}
-
-		public void SetCustomAttribute (params object[] args)
+		public static void DefineManifestResource (this ModuleBuilder builder, params object[] args)
 		{
 			throw new NotSupportedException ();
 		}
 	}
 
-	public class EventBuilder
+	static class TypeBuilderExtensions
 	{
-		public void SetAddOnMethod (MethodBuilder mdBuilder)
+		public static void AddDeclarativeSecurity (this TypeBuilder builder, params object[] args)
 		{
 			throw new NotSupportedException ();
 		}
 
-		public void SetRemoveOnMethod (MethodBuilder mdBuilder)
-		{
-			throw new NotSupportedException ();
-		}
-
-		public void SetCustomAttribute (params object[] args)
+		public static Type CreateType (this TypeBuilder builder)
 		{
 			throw new NotSupportedException ();
 		}
 	}
-
-}
\ No newline at end of file
+}
```

Comment:
While you're modifying this file I think we should rename it from monotouch.cs to something else, since we use it in the winaot, orbis and testing_aot_full profiles too now.